### PR TITLE
[Java] Support starting named actors in different namespace.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
Allow you start actors in different namespace instead of the driver namespace.
Usage is simple:
```java
Ray.init(namespace="a");
/// Named actor a will starts in namespace `b`
ActorHandle<A> a = Ray.actor(A::new).setName("myActor", "b").remote();
```

## Related issue number
#22563 #25665

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
@shrekris-anyscale @simon-mo

**Review by simon-mo — APPROVED** [review]

File: java/api/src/main/java/io/ray/api/call/ActorCreator.java [hunk 1]
```diff
@@ -60,4 +60,10 @@ public ActorCreator<A> setRuntimeEnv(RuntimeEnv runtimeEnv) {
     builder.setRuntimeEnv(runtimeEnv);
     return this;
   }
+
+  /** Set the namespace that this actor uses. */
+  public ActorCreator<A> setNamespace(String namespace) {
```
Comment (raulchen) [inline java/api/src/main/java/io/ray/api/call/ActorCreator.java:65]: Should the namespace always set along with a name? 
If yes, we should change the API to `setName(name, namespace)`
The semantic seems unclear if only setting namespace without name. Not Sure if Python allows this.
@simon-mo @jovany-wang

Comment (kfstorm) [inline java/api/src/main/java/io/ray/api/call/ActorCreator.java:65]: Why not put it in `BaseActorCreator` to support setting namespace for Python actors as well?

Comment (simon-mo) [reply java/api/src/main/java/io/ray/api/call/ActorCreator.java:65]: yes namespace is pretty coupled with name

File: java/api/src/main/java/io/ray/api/call/BaseActorCreator.java [hunk 2]
```diff
@@ -25,6 +25,19 @@ public T setName(String name) {
     return self();
   }
 
+  /**
+   * Set the actor name of a named actor.
+   *
+   * @param name The name of the named actor.
+   * @param namespace The namespace that this actor will live.
```
Comment (raulchen) [inline java/api/src/main/java/io/ray/api/call/BaseActorCreator.java:32]:
```suggestion
   * Set the actor name along with a different namespace.
   *
   * @param name The name of the named actor.
   * @param namespace The namespace that this actor will live in.
```

**Review by raulchen — APPROVED** [review]

**Review by kfstorm — APPROVED** [review]

**Pull request merged**